src/org/daisy/dotify/formatter/impl/row/SegmentProcessor.java

```diff
@@ -116,6 +119,7 @@
         this.groupMarkers = new ArrayList<>(template.groupMarkers);
         this.groupIdentifiers = new ArrayList<>(template.groupIdentifiers);
         this.leaderManager = new LeaderManager(template.leaderManager);
+        this.externalReference = template.externalReference;
```

Comment:
Why at most one?
I'd expect here `this.externalReferences = new ArrayList<>(template.externalReference);`

Reply:
An external reference object is an object that could contain multiple values. Later, when we want to implement Berts' more extensive use-case, we might need more than one. But in this case, where we add attributes to a row, there can only be one object that can contain whatever you like. In this implementation, it is a map, but you can use the API to implement something else.

Reply:
Can you have more than one `external reference` objects in a single row?

Reply:
No, you can only have one external reference object per block in my implementation.

Reply:
>No, you can only have one external reference object per block in my implementation.

Please see my remark below at `case ExternalReference:`